basic/src/test/java/org/jboss/hal/testsuite/test/configuration/undertow/UndertowOperations.java

```diff
@@ -139,153 +73,115 @@ public void removeSecurityRealm(String realmName) throws CommandFailedException,
+        if (keepReference) {
+            usedWorkers.add(worker);
         }
+        return worker;
     }
 
-    public String createHTTPServer() throws InterruptedException, IOException, TimeoutException {
-        String name = "HTTPServer_" + RandomStringUtils.randomAlphanumeric(6);
-        ResourceAddress address = httpServerTemplate.resolve(context, name);
-        executeAddAction(address, true);
-        operations.readResource(Address.of("subsystem", "undertow").and("server", name)); //Workaround for situation when server is listed as unavailable dependency
-        return name;
-    }
-
-    public void removeHTTPServer(String httpServer) throws InterruptedException, TimeoutException, IOException {
-        executeRemoveAction(httpServerTemplate.resolve(context, httpServer));
+    public void removeWorker(String name) throws IOException, OperationException {
+        Address address = Address.subsystem("io").and("worker", name);
+        operations.removeIfExists(address);
     }
 
-    public String createServletContainer() throws InterruptedException, TimeoutException, IOException {
-        String name = "ServletContainer_" + RandomStringUtils.randomAlphanumeric(6);
-        ResourceAddress address = servletContainerTemplate.resolve(context, name);
-        executeAddAction(address);
+    public String createBufferPool() throws IOException {
+        String name = "UndertowBufferPool_" + RandomStringUtils.randomAlphanumeric(6);
+        Address address = Address.subsystem("io").and("buffer-pool", name);
+        operations.add(address);
         return name;
     }
 
-    public void removeServletContainer(String servletContainer) throws InterruptedException, TimeoutException, IOException {
-        executeRemoveAction(servletContainerTemplate.resolve(context, servletContainer));
-    }
-
-    private void executeRemoveAction(ResourceAddress address) throws InterruptedException, IOException, TimeoutException {
-        Dispatcher dispatcher = new Dispatcher();
-        dispatcher.execute(new Operation.Builder("remove", address).build()); //TODO add creaper to all methods, then remove
-        dispatcher.close();
-        reloadIfRequiredAndWaitForRunning();
-    }
-
-    private boolean executeReadResourceAction(ResourceAddress address) {
-        Dispatcher dispatcher = new Dispatcher();
-        boolean result = dispatcher.execute(new Operation.Builder("read-resource", address).build()).isSuccessful();
-        dispatcher.close();
-        return result;
-    }
-
-    private void executeAddAction(ResourceAddress address, Map<String, String> params, boolean reload) throws InterruptedException, IOException, TimeoutException {
-        Operation.Builder operationBuilder = new Operation.Builder("add", address);
-        for (Map.Entry<String, String> entry : params.entrySet()) {
-            operationBuilder.param(entry.getKey(), entry.getValue());
+    public String createBufferPool(boolean keepReference) throws IOException {
```

Comment:
Is `keepReference` argument necessary? Dou you really expect false value to be passed?